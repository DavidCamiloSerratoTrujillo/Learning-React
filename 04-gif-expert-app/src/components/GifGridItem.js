import React from 'react';
import PropTypes from 'prop-types'
export const GifGridItem = ({title,url}) => {
  return (
    <div className='card cola cao'>
        <img src ={url} alt ={title}/>{/*Se cargan las imagenes*/}
        <p>{title}</p> 
    </div>
  )
}

GifGridItem.propTypes ={
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}