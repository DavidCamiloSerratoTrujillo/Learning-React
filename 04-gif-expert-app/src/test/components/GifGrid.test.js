import {shallow} from 'enzyme';
import {GifGrid}from '../../components/GifGrid'
import React from 'react';
import {useFetchGifs} from '../../hooks/useFectchGifs'
jest.mock('../../hooks/useFectchGifs');

describe('Pruebas a componente gifGrid',()=>{
    useFetchGifs.mockReturn({
        data:[],
        loading: true
    })

    const categories = 'one push';
    let wrapper = shallow(<GifGrid categories={categories}/>)

    // beforeEach(()=>{
       
    // })

    test('Hacer match en el primer snapshot',()=>{

      

        wrapper = shallow(<GifGrid categories={categories}/>)
        expect(wrapper).toMatchSnapshot();
        
    })

    // test('debe de mostrar items cuando se cargan imagenes',()=>{

    // })
});