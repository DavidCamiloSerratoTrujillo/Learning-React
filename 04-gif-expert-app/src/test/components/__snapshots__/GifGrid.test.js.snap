// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a componente gifGrid Hacer match en el primer snapshot 1`] = `
<Fragment>
  <h3 />
  <p>
    Loading
  </p>
  <div
    className="card-grid "
  />
</Fragment>
`;
